frontend/packages/dev-console/src/components/import/devfile/devfileHooks.ts

```diff
@@ -21,24 +120,33 @@ export const useDevfileServer = (
 
   const {
     name,
-    git: { url, ref, dir },
+    git: { url, ref, dir, type, secretResource },
     devfile,
   } = values;
   const smartSlashDir = prefixDotSlash(suffixSlash(dir));
 
   const { devfileContent, devfilePath } = devfile || {};
 
-  const devfileData = React.useMemo(() => {
+  const devfileDataPromise = React.useMemo(async () => {
```

Comment:
Works fine, but really uncommon :man_shrugging:

Reply:
I didnt get it. Returning a promise from useMemo is uncommon or?